### PR TITLE
Support teams in extended package syntax and add team tests

### PR DESCRIPTION
**Pull request by dimaryaz**
(Most of this was written by kurlov and already reviewed by me - but I rewrote the regex parsing.)

### CODE REVIEW

File: compiler/quilt/tools/command.py [hunk 1]
```diff
@@ -908,10 +895,7 @@ def install_via_requirements(requirements_str, force=False):
     else:
         yaml_data = yaml.load(requirements_str)
     for pkginfo in yaml_data['packages']:
-        owner, pkg, subpath, hash, version, tag = parse_package_extended(pkginfo)
-        package = owner + '/' + pkg
-        if subpath:
-            package += '/' + "/".join(subpath)
+        package, hash, version, tag = parse_package_extended(pkginfo)
```
Comment (kevinemoore) [inline compiler/quilt/tools/command.py]: Since you're changing the line already, avoid using 'hash'?

File: compiler/quilt/tools/command.py [hunk 2]
```diff
@@ -80,24 +83,15 @@ class CommandException(Exception):
...
+    """
+    Parses the extended package syntax and returns a tuple of (package, hash, version, tag).
+    """
+    match = EXTENDED_PACKAGE_RE.match(name)
+    if match is None:
+        pkg_format = '[team:]owner/package_name/path[:v:<version> or :t:<tag> or :h:<hash>]'
         raise CommandException("Specify package as %s." % pkg_format)
-    return package, hash, version, tag
+
+    return match.groups()
```
Comment (Kurlov) [inline compiler/quilt/tools/command.py]: oh, that's really much better.

File: compiler/quilt/test/test_install.py [hunk 3]
```diff
@@ -76,8 +78,7 @@ def test_install_latest(self):
         command.install('foo/bar')
         teststore = PackageStore(self._store_dir)
 
-        with open(os.path.join(teststore.package_path(None, 'foo', 'bar'),
-                               Package.CONTENTS_DIR,
+        with open(os.path.join(teststore.package_path(None, 'foo', 'bar'), Package.CONTENTS_DIR,
```
Comment (kevinemoore) [inline compiler/quilt/test/test_install.py]: Not related to this PR, but should we have a helper for the contents directory? Or maybe, package_path should return the path to the contents directory?

Comment (Kurlov) [reply compiler/quilt/test/test_install.py]: Yes, there is a helpers for file validation. I will create a separate PR to clean these tests a little along with new test for installing team based dependencies from .yml

File: compiler/quilt/tools/command.py (hunk 1 above)
Comment (Kurlov) [reply compiler/quilt/tools/command.py]: I am sorry, do you mean to prevent override built-in function?

**Pull request merged**